### DISCUSSION TITLE
Administer Multiple Doses in R

### DISCUSSION THREAD

**Discussion by DonkeySaddle** [Open-Systems-Pharmacology/Forum#1644] [How-Tos] [answered]
How can I manipulate a simulation in R so that I can administer multiple doses for a given drug using the R ospsuite package?

For example, I want to administer a dose of 5mg and 10mg of Drug A and drug B respectively, at time 0, and 5mg of drug A at 30 minutes, and 7 mg of drug B at 60 minutes.

I couldn't figure out how to do it [here]( https://www.open-systems-pharmacology.org/OSPSuite-R/articles/set-values.html). 

Any help would be much appreciated. 

Thanks,

DS

**Comment by wilbertdew** [comment]
Hi @DonkeySaddle,

I believe its not possible to add dosing events in R yet. What you can do is add the number of dosing events that you maximally want to apply in your simulations to your model in PK-Sim/MoBi, and set the dose of some of these dosing events to 0 in R if you want to simulate less dosing events.

**Reply by DonkeySaddle** [reply]
@wilbertdew Thank you for your response, this seems like a great hack! Appreciate it. 

So let's say in my PK-Sim/Mobi simulation have created an administration protocol for two doses of Itraconazole spaced 12 hours apart. I won't be able to change the times I administer the drug, but I could set the first dose to 250mg and the second one to 0mg. Would this be the code in R (I set the values equal to a list containing 250 and 0): 

```
#Create a path and get the parameters of Itraconazole
ItracanDoseParamPath <- "Applications|PO SD 100 mg|Capsule fed Itraconazole|Application_1|ProtocolSchemaItem|Dose"
ItracanDoseParam <- getParameter(ItracanDoseParamPath, sim)

#Set doses to 250mg and 0mg  
Itracan_val <- toBaseUnit(quantity = ItracanDoseParam, values = list(250, 0), unit = "mg") 
setParameterValues(parameters = ItracanDoseParam, values = Itracan_val)

```

Thanks again!

DS

**Reply by DonkeySaddle** [reply]
Okay, I have played around with the R code a bit, what I posted above is way off, but I am not sure what the fix is. Will share if I find a solution. 

Any help would be greatly appreciated. 

Thanks, 

DS

**Reply by PavelBal** [reply]
Your code looks fine, what is the problem you are having? You can also take a look at the `setParameterValuesByPath()` function, where you can directly pass the unit your values are in.

**Reply by DonkeySaddle** [reply]
I am getting this error: 

> setParameterValues(parameters = ItracanDoseParam, values = Itracan_val)

```
Error in validateIsSameLength(quantities, values) : 
  setParameterValues: Arguments 'quantities, values' must have the same length, but they don't!

```
I will look into the `setParameterValuesByPath()` function and report back. Thanks @PavelBal.

**Reply by msevestre** [reply]
The error message explains the problem for once. You are setting two values for one parameter. This doesn't work. You need to either set two values for two parameters or one value for one parameter. 
And once you set the values, you need to run your simulation in order to get updated results

**Reply by DonkeySaddle** [reply]
Right, I figured that much out, just am not sure how to set two values for two parameters in this case. The parameters in this case being the doses (which are repeated) and the values being the respective dose amounts.

**Reply by msevestre** [reply]
There are one parameter for each dose. 
So get them by path (Applications|PO SD 100 mg|Capsule fed Itraconazole|Application_1|ProtocolSchemaItem|Dose) here it says applciaiton_1 ....

See the documentation on retrieving entities by path. You can use wildcard etc..so that it will me much easier to update if you go to 3 applications.

**Reply by DonkeySaddle** [reply]
Ahhh! Okay, now I feel stupid. Thanks @msevestre, can’t test it now, but that should solve it…

Really appreciate it!  
DS

**Reply by msevestre** [reply, marked as answer]
no problem. 
see
https://www.open-systems-pharmacology.org/OSPSuite-R/reference/getAllParametersMatching.html

and 

see https://www.open-systems-pharmacology.org/OSPSuite-R/articles/set-values.html




```
# Get objects representing the molecule Aciclovir in all containers
allAciclovirMolecules <- getAllMoleculesMatching("Organism|**|Aciclovir", sim)

# Set initial values to 10 µmol in all containers
setMoleculeInitialValues(allAciclovirMolecules, rep(10, length(allAciclovirMolecules)))
```

the same methods exist for parameters. You get an array when using wildcard. Very very cool

**Reply by DonkeySaddle** [reply]
@msevestre, this was incredibly helpful and, indeed, very very cool. Makes working in R much easier and more powerful. 

Thanks again.

DS
